kotlinKapt crash in some cases when use ArrayList<T> or ArrayDeque<T> as return type in controller

### Expected Behavior

Project should build successfully.

### Actual Behaviour

After calling `gradlew clean build` I've got following error:
```
> Task :kaptKotlin
Note: Generating OpenAPI Documentation
error: Error finalizing type visitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor@107268a5]: null
Note: Creating bean classes for 2 type elements
error: Error finalizing type visitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor@107268a5]: null

> Task :kaptKotlin FAILED

FAILURE: Build failed with an exception.
```

### Steps To Reproduce

I've created new project in Micronaut Launcher https://micronaut.io/launch/ with following options:

Application Type: Micronaut Application
Java version: 17
Micronaut Version: 3.8.0
Language: Kotlin
Build Tool: Gradle Kotlin
Test Framework: Kotest
In features: swagger-ui and openapi

Next I've created new simple data class:

`data class AC(val someField: String)`

and changed index method so it returns empty `ArrayList<AB>`

`fun index() = ArrayList<AB>()`

Then I call  `gradlew clean build` and it fails.

What is funny following class names works:

```
A
Ab
SomeClass
Xbcd
```

but following class names are not:
```
Abc
AB
ABc
AwesomeClass
```

what is more consusing is that it always works for List or MutableList but doesn't work for ArrayList and ArrayDeque

It starts to crash from version 4.8.0 of micronaut-openapi, after downgrading to 4.7.1 it works like a charm.

Full gradle stacktrace in attachment.
[fullstacktrace.txt](https://github.com/micronaut-projects/micronaut-openapi/files/10401018/fullstacktrace.txt)


### Environment Information

- JDK 17
- Gradle wrapper 7.5.1

### Example Application

https://github.com/marendil/demo

### Version

3.8.0

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
@@ -844,7 +844,7 @@ abstract class AbstractOpenApiVisitor {
                         } else {
                             schema = getPrimitiveType(Object.class.getName());
                         }
-                        List<FieldElement> fields = type.getFields();
+                        List<FieldElement> fields = type.getPackageName().startsWith("java.util") ? Collections.emptyList() : type.getFields();
                         if (schema != null && fields.isEmpty()) {
                             schema = SchemaUtils.arraySchema(schema);
                         } else {

--- a/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerRequiresVisitorSpec.groovy
+++ b/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerRequiresVisitorSpec.groovy
@@ -25,7 +25,7 @@ import io.micronaut.http.annotation.Post;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -35,7 +35,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -75,7 +75,7 @@ import io.micronaut.http.annotation.Post;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -85,7 +85,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -130,7 +130,7 @@ import java.util.List;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -140,7 +140,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -185,7 +185,7 @@ import java.util.List;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -195,7 +195,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -205,7 +205,7 @@ class Controller2 {
 class Controller3 {
 
     @Post("/c3")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -251,7 +251,7 @@ import java.util.List;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -261,7 +261,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -308,7 +308,7 @@ import java.util.List;
 class Controller1 {
 
     @Post("/c1")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -319,7 +319,7 @@ class Controller1 {
 class Controller2 {
 
     @Post("/c2")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }
@@ -330,7 +330,7 @@ class Controller2 {
 class Controller3 {
 
     @Post("/c3")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 }

--- a/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerVisitorSpec.groovy
+++ b/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerVisitorSpec.groovy
@@ -6,6 +6,7 @@ import io.swagger.v3.oas.models.OpenAPI
 import io.swagger.v3.oas.models.Operation
 import io.swagger.v3.oas.models.PathItem
 import io.swagger.v3.oas.models.Paths
+import io.swagger.v3.oas.models.media.Schema
 import spock.lang.Issue
 
 class OpenApiControllerVisitorSpec extends AbstractOpenApiTypeElementSpec {
@@ -62,6 +63,84 @@ class MyBean {}
         paths.containsKey("/five")
     }
 
+    void "test java.util collections"() {
+
+        given:
+        buildBeanDefinition('test.MyBean', '''
+package test;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Post;
+
+@Controller
+class HelloWorldController {
+
+    @Post("/ArrayList")
+    public ArrayList<AB> ArrayList() {
+        return null;
+    }
+
+    @Post("/ArrayDeque")
+    public ArrayDeque<AB> ArrayDeque() {
+        return null;
+    }
+
+    @Post("/LinkedBlockingDeque")
+    public LinkedBlockingDeque<AB> LinkedBlockingDeque() {
+        return null;
+    }
+
+    @Post("/SynchronousQueue")
+    public SynchronousQueue<AB> SynchronousQueue() {
+        return null;
+    }
+}
+
+class AB {
+    public String someField;
+}
+
+@jakarta.inject.Singleton
+class MyBean {}
+''')
+        when:
+        Paths paths = Utils.testReference?.paths
+        Map<String, Schema> schemas = Utils.testReference?.components?.schemas
+
+        then:
+        paths
+        paths.size() == 4
+
+        paths."/ArrayList"
+        paths."/ArrayList".post.operationId == 'ArrayList'
+        paths."/ArrayList".post.responses."200".content."application/json".schema.type == 'array'
+        paths."/ArrayList".post.responses."200".content."application/json".schema.items.$ref == '#/components/schemas/AB'
+
+        paths."/ArrayDeque"
+        paths."/ArrayDeque".post.operationId == 'ArrayDeque'
+        paths."/ArrayDeque".post.responses."200".content."application/json".schema.type == 'array'
+        paths."/ArrayDeque".post.responses."200".content."application/json".schema.items.$ref == '#/components/schemas/AB'
+
+        paths."/LinkedBlockingDeque"
+        paths."/LinkedBlockingDeque".post.operationId == 'LinkedBlockingDeque'
+        paths."/LinkedBlockingDeque".post.responses."200".content."application/json".schema.type == 'array'
+        paths."/LinkedBlockingDeque".post.responses."200".content."application/json".schema.items.$ref == '#/components/schemas/AB'
+
+        paths."/SynchronousQueue"
+        paths."/SynchronousQueue".post.operationId == 'SynchronousQueue'
+        paths."/SynchronousQueue".post.responses."200".content."application/json".schema.type == 'array'
+        paths."/SynchronousQueue".post.responses."200".content."application/json".schema.items.$ref == '#/components/schemas/AB'
+
+        schemas
+        schemas.size() == 1
+
+        schemas.AB.type == 'object'
+        schemas.AB.properties.someField.type == 'string'
+    }
+
     void "test hidden endpoint"() {
 
         given:
@@ -82,19 +161,19 @@ import java.util.List;
 class HelloWorldController {
 
     @Post("/public")
-    public HttpResponse<String> publicEnpoint() {
+    public HttpResponse<String> publicEndpoint() {
         return null;
     }
 
     @Hidden
     @Post("/private")
-    public HttpResponse<String> internalEnpoint() {
+    public HttpResponse<String> internalEndpoint() {
         return null;
     }
 
     @Operation(hidden = true)
     @Post("/private-operation")
-    public HttpResponse<String> internalEnpoint2() {
+    public HttpResponse<String> internalEndpoint2() {
         return null;
     }
 

--- a/src/main/docs/guide/endpoints/endpointstags.adoc
+++ b/src/main/docs/guide/endpoints/endpointstags.adoc
@@ -2,5 +2,5 @@ You can also provide some tags for all endpoints with the `endpoints.tags=<comma
 
 .openapi.properties
 ----
-endpoints.tags=Management Enpoints
+endpoints.tags=Management Endpoints
 ----

